### PR TITLE
new: add ability to retrieve field name via BSONFieldName

### PR DESCRIPTION
**Pull request by aaslamin**
This PR consumes the functionality to translate the custom BSON field name added to the spec here: https://github.com/aporeto-inc/elemental/pull/76/commits

### CODE REVIEW

**Review by primalmotion — CHANGES_REQUESTED** [review]
Needs tests on options and actual compiler test

File: manipmongo/compiler.go [hunk 1]
```diff
@@ -151,14 +133,14 @@ func CompileFilter(f *elemental.Filter) bson.D {
...
-				subs = append(subs, CompileFilter(sub))
+				subs = append(subs, CompileFilter(sub, opts...))
 			}
 			ands = append(ands, bson.D{{Name: "$and", Value: subs}})
 
 		case elemental.OrFilterOperator:
 			subs := []bson.D{}
 			for _, sub := range f.OrFilters()[i] {
-				subs = append(subs, CompileFilter(sub))
+				subs = append(subs, CompileFilter(sub, opts...))
```
Comment (aaslamin) [inline manipmongo/compiler.go:143]: The new unit tests I added cover the recursive cases to ensure the configured options are passed down on recursive calls 🔁 

@primalmotion 👀

**Comment by aaslamin** [comment]
### Update

• 💯% test coverage has been added for the new `CompilerOption` along with end-to-end tests demonstrating it in action.

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "victoria-metrics",
    "component": "manipulate",
    "pr-id": "127",
    "commit-sha": "bf22db4122e4dcf62c73a2c107c8fe3d3c3cad72",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "backend",
    "pr-id": "726",
    "commit-sha": "fe450be2aee3ffa59e1fe9d6a2d1ed4958c46e5e",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "gaia",
    "pr-id": "471",
    "commit-sha": "8c87f216ac3978968e3d2ff087a3e4c60b60ae1f",
    "pipeline": "master"
  },
  {
    "project": "victoria-metrics",
    "component": "underwater",
    "pr-id": "104",
    "commit-sha": "52843e51e546536d79d893dffb53428d30cf8c5d",
    "pipeline": "master"
  }
]
```

**Review by primalmotion — APPROVED** [review]

**Pull request merged**